lib/interactor/organized_interactor.rb

```diff
@@ -0,0 +1,40 @@
+module Interactor
+  # Internal: this class allows us to run interactors with regard of given
+  # options, such as :if, :unless, :before, :after
+  class OrganizedInteractor
+    attr_reader :interactor, :options
+
+    def initialize(interactor, options = {})
+      @interactor = interactor
+      @options = options
+    end
+
+    def call!(context, within_organizer)
```

Comment:
I'd rather use keyword argument for `within_organizer` here, but currently gemspec is targeted to Ruby 2.0.